### ISSUE TITLE
Assertion is fired at the start of debugging session

### ISSUE THREAD

**Issue by longagofaraway** [Cockatrice/Cockatrice#4048]
<b>System Information:</b>
Client Version: 2.7.4-custom(7ab561f) (2020-07-17)
Client Operating System: Windows 10 (10.0)
Build Architecture: 64-bit
Qt Version: 5.12.5
  

  
_______________________________________________________________________________________
At the start of Cockatrice debug session, when cards from database are being inserted into the main screen, assertion is triggered.
![assert](https://user-images.githubusercontent.com/19328424/87820378-a8afa180-c876-11ea-9044-0269cd9264ae.png)

from this call https://github.com/Cockatrice/Cockatrice/blob/master/cockatrice/src/carddatabasemodel.cpp#L130
Assertion is triggered on second or third card. Release build is still working for me.

What helped me is: I changed [CardDatabaseDisplayModel::rowCount](https://github.com/Cockatrice/Cockatrice/blob/master/cockatrice/src/carddatabasemodel.cpp#L177) from
`return qMin(QSortFilterProxyModel::rowCount(parent), loadedRowCount);`
to
```
if (parent.isValid())
        return 0;

return QSortFilterProxyModel::rowCount(parent);
```
The parent check seems in accordance with https://doc.qt.io/qt-5/qabstractitemmodel.html#rowCount
And seems like loadedRowCount is not updating in time, so I ignored it. But I don't know what I'm doing, not familiar with Qt.

[log.txt](https://github.com/Cockatrice/Cockatrice/files/4939667/log.txt)
[stacktrace.txt](https://github.com/Cockatrice/Cockatrice/files/4939670/stacktrace.txt)

**Comment by ebbit1q** [comment]
This seems to be exclusive to the ms visual studio ide, while it most likely does not affect functionality at all it'd be nice to not have this innocuous assert trigger.

**Comment by ebbit1q** [comment]
@longagofaraway would you mind putting your changes in a pull request for review?

**Comment by longagofaraway** [comment]
> would you mind putting your changes in a pull request for review?

I don't think this is the right fix, I'm still getting assertions and I don't understand what's happening there. You still think I need to present it as a pull request?

**Comment by ebbit1q** [comment]
you're getting other assertions that fail or is it just this?

**Comment by longagofaraway** [comment]
If I apply a suggested fix from my initial post there may be other assertions, but I can just ignore them and continue debugging. Without this fix there are recursive assertions and that just end badly.

**Comment by ebbit1q** [comment]
Are you sure this is not a bug within the ide/compiler itself?

**Comment by longagofaraway** [comment]
Strange thing are happening, how can one be sure of anything. This issue might need a whole debugging session with qt sources.

**Comment by ZeldaZach** [comment]
cc @ctrlaltca this seems like a problem that can impact new developers.
